### PR TITLE
fix: merge extra params with snippet extra params

### PR DESCRIPTION
**Pull request by tiborux**
With this approach instead of using multiple ExtraParams component, only one SnippetExtraParams component will be used. So to test it, just launch the server and see how the extra params are loaded and emitted correctly, you can try to add more values at home.vue in:

` protected initialExtraParams = { store: 'Portugal' };`

to check that you can add more params and they will be combined with snippet one.

EX-5205

### CODE REVIEW

**Review by diegopf — DISMISSED** [review]

File: packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue [hunk 1]
```diff
@@ -65,7 +73,7 @@
       currency,
       ...snippetExtraParams
     }: SnippetConfig): void {
-      forEach(snippetExtraParams, (name, value) => {
+      forEach({ ...snippetExtraParams, ...this.values }, (name, value) => {
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue:76]: Where is the test for this new feature? 🤣

File: packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue [hunk 2]
```diff
@@ -65,7 +73,7 @@
...
         if (this.notAllowedExtraParams.includes(name)) {
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue:77]: Just noticed that we are using both destructuring and an array named `notAllowedExtraParams` for discarding parameters, I think we should pick one of those options. We can do it in another PR if you prefer so. 
Also, `scope` parameter is discarded, and its solely purpose is to be sent to the API to group requests depending on the device. We can do it in another PR if you wish so.

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue (hunk 1 above)
Comment (tiborux) [reply packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue:76]: the tests are the parents

**Review by diegopf — APPROVED** [review]

**Comment by diegopf** [comment]
I see the build is failing but looks like it is due to the tagging flaky tests we currently have 😓

**Review by javieri-empathy — APPROVED** [review]
